### PR TITLE
Standardize usage of AWS and Serilog using ITextFormatter.

### PR DESCRIPTION
**Pull request by wparad**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -36,8 +36,7 @@ CloudWatchSinkOptions options = new CloudWatchSinkOptions
 };
 
 // setup AWS CloudWatch client
-AWSCredentials credentials = new BasicAWSCredentials(myAwsAccessKey, myAwsSecretKey);
-IAmazonCloudWatchLogs client = new AmazonCloudWatchLogsClient(credentials, myAwsRegion);
+var client = new AmazonCloudWatchLogsClient(myAwsRegion);
```
Comment (thoean) [inline readme.md]: Let's add a comment to the `Configuration` section in this readme so it's not ignored. There are many options and choices to be made, and that's not part of this library. I suggest to be very clear in the readme about this.

File: src/Serilog.Sinks.AwsCloudWatch/AwsCloudWatchConfigurationExtension.cs [hunk 2]
```diff
@@ -39,319 +39,44 @@ public static LoggerConfiguration AmazonCloudWatch(this LoggerSinkConfiguration
...
+        /// <exception cref="ArgumentNullException"><paramref name="formatter"/> is <see langword="null"/>.</exception>
         public static LoggerConfiguration AmazonCloudWatch(
             this LoggerSinkConfiguration loggerConfiguration,
             string logGroupName,
             ITextFormatter formatter,
-            string accessKey,
-            string secretAccessKey,
-            string regionName = null,
-            string logStreamNamePrefix = null,
+            ILogStreamNameProvider logStreamNameProvider = null,
```
Comment (thoean) [inline src/Serilog.Sinks.AwsCloudWatch/AwsCloudWatchConfigurationExtension.cs]: There's this and a 2nd option to configure this sink. Can we:
* Be explicit why these 2 exist? More or less, how do we prevent to be swamped with pull requests that will end up with the same as in v3?
* Can we even reduce it to a single one?
* Does this support the appsettings based Configuration for Serilog? If not, should we be explicit on why we're not supporting it? Many seem to be wanting it, is it easier to just provide it out of the box, potentially with a full example?

File: src/Serilog.Sinks.AwsCloudWatch/CloudWatchLogSink.cs [hunk 3]
```diff
@@ -52,7 +53,7 @@ public class CloudWatchLogSink : PeriodicBatchingSink
         private bool hasInit;
         private string logStreamName;
         private string nextSequenceToken;
-        private readonly ILogEventRenderer renderer;
+        private readonly ITextFormatter textFormatter;
```
Comment (thoean) [inline src/Serilog.Sinks.AwsCloudWatch/CloudWatchLogSink.cs]: After merging, we'll need to update https://github.com/Cimpress-MCP/serilog-sinks-awscloudwatch/wiki/Using-a-JSON-renderer

File: readme.md (hunk 1 above)
Comment (wparad) [reply readme.md]: I'm not following what you are asking.

Comment (wparad) [reply readme.md]: Cleaned up documentation.

File: src/Serilog.Sinks.AwsCloudWatch/AwsCloudWatchConfigurationExtension.cs (hunk 2 above)
Comment (wparad) [reply src/Serilog.Sinks.AwsCloudWatch/AwsCloudWatchConfigurationExtension.cs]: Reduced to only one extension, the dynamic construction of parameters can be done with two alternative recommendation libraries.

File: src/Serilog.Sinks.AwsCloudWatch/CloudWatchLogSink.cs (hunk 3 above)
Comment (wparad) [reply src/Serilog.Sinks.AwsCloudWatch/CloudWatchLogSink.cs]: Done?

Comment (thoean) [reply src/Serilog.Sinks.AwsCloudWatch/CloudWatchLogSink.cs]: https://github.com/Cimpress-MCP/serilog-sinks-awscloudwatch/wiki/Using-a-JSON-formatter might need some tweaking, like the interface it derives from doesn't exist anymore. That said, we can follow up later.

File: readme.md [hunk 4]
```diff
@@ -4,69 +4,97 @@ This Serilog Sink allows to log to [AWS CloudWatch](https://aws.amazon.com/cloud
...
+  var client = new AmazonCloudWatchLogsClient(myAwsRegion);
+
+  // Attach the sink to the logger configuration
+  Log.Logger = new LoggerConfiguration()
+    .WriteTo.AmazonCloudWatch(options, client)
+    .CreateLogger();
+  ```
+
+##### Configuration via config file
+While not recommended, it is still possible to config the library via a configuration file.  There are two libraries which provide these capabilities.
```
Comment (thoean) [inline readme.md]: `to config the library via a configuration file` ==> `to configure`

That said, I like the readme file a lot better than what we had before.

**Pull request merged**